exa/core/box.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) 2015-2020, Exa Analytics Development Team
+# Distributed under the terms of the Apache License 2.0
+"""
+Data container
+####################################
+An in-memory related data container.
+"""
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import networkx as nx
+from traitlets import Unicode, Dict
+
+import exa
+from exa import Base
+from exa.core.data import Data
+
+class Box(Base):
+    name = Unicode(help="string name of container")
+    meta = Dict(help="metadata dictionary")
+    description = Unicode(help="container description")
+
+    def copy(self, **kws):
+        return self.__class__(**kws)
+
+    def slice(self, key):
+        pass
+
+    def groupby(self):
+        pass
+
+    def info(self):
+        pass
+
+    def memory(self):
+        pass
+
+    def network(self):
+        pass
+
+    def save(self, name=None, directory=None, parquet_backend='pyarrow'):
+        """Save a bundle of files to a tarball."""
```

Comment:
Says tarball but looks like saving to flat files in directory on line 49. I think the tarball idea is actually worth doing. Is it possible to write the `data.save` buffer directly into a `tar = tarfile.open(bundle, ...)` object? Or does one first have to `data.save(bundle)` then add that file (on disk) to the archive, then save the archive (this would create 2x amount of data on disk temporarily?)?

Reply:
I like the idea of constructing the tarball directly for the container. `Data.save` should support saving to the `tarfile` buffer of the container. It should also be easy to save and manage a `Data` object independently, though it does seem a bit disappointing to require 2 files of different formats.